haiku-games/hexvexed/hexvexed-0.0.1.recipe

```diff
@@ -0,0 +1,51 @@
+SUMMARY="A tile puzzle game for Haiku"
+DESCRIPTION=""HexVexed is a maddening addictive and easy to play puzzle game:
+Arrange hexagonal tiles into a grid so that the numbers and colours match
+where they touch. This game is a version DarkWyrm's BeVexed, but instead
+of squares, the tiles a hexagonal shaped. It's 1.5x the fun!
+
+HOMEPAGE="https://bitbucket.org/scottmc/hexvexed"
+COPYRIGHT="2009-2016 Scott McCreary"
+LICENSE="MIT"
+REVISION="1"
+commit="68437a22c788"
+SOURCE_URI="$HOMEPAGE/get/$commit.zip"
+CHECKSUM_SHA256="35e0904d91f15fefe801cdb7b9f9bdf4dc925de69f5b8818bec24cc4c22dc6d7"
+SOURCE_DIR="scottmc-hexvexed-$commit"
+
+ARCHITECTURES="x86_gcc2 x86 ?x86_64"
+
+PROVIDES="
+	hexvexed = $portVersion
+	app:HexVexed = $portVersion
+	"
+REQUIRES="
+	haiku
+	"
+	
+BUILD_REQUIRES="
+	haiku_devel
+	"
+BUILD_PREREQUIRES="
+	makefile_engine
+	cmd:make
+	cmd:g++
+	"
+
+BUILD()
+{
+	make $jobArgs OBJ_DIR=objects
+	make $jobArgs OBJ_DIR=objects bindcatalogs
+}
+
+INSTALL()
+{
+	mkdir -p $appsDir
+	mkdir -p backgrounds
+	mkdir -p $dataDir/HexVexed
+
+	cp -r backgrounds $dataDir/HexVexed
+      
+	cp -a objects/HexVexed $appsDir
+        addAppDeskbarSymlink $appsDir/HexVexed
```

Comment:
The 555 is the mode or permissions (c.f.the chmod command). Here we are installing it with r-xr-xr-x permissions.

BTW, if you see a "-c" (or -C) in some of the examples, then you should skip that option. It is completely useless in a INSTALL function of a recipe.

> How to copy a file using install is it "install -d-m 555 -s hexvexed "

No. "install -d" will only _install_ (that is, _create_) directories. (So that's what replaces the "mkdir -p".)
I think we can use the following INSTALL for this recipe:
```
INSTALL()
{
	install -d -m 755 $appsDir $dataDir/HexVexed/backgrounds
	install -t $appsDir -m 555 -s objects/HexVexed
	addAppDeskbarSymlink $appsDir/HexVexed
}
```
I haven't tested this yet, however.
But as you can see, it looks much nicer than using mkdir and cp, isn't it?